src/controllers/blockModel.py

```diff
@@ -19,3 +24,94 @@ def find_block(self, x, y, z, collection):
         if (block != None):
             return Block(block.keys(), block.values())
         return False 
+    
+    def reblock_model_attributes(self, collection):
+        block = collection.find({})[0]
+        self.model_keys = list(block.keys())[5:]
+        return(list(block.keys())[5:])
+
+    def reblock(self,collection, rx, ry, rz, attributes_types, mass_attribute):
+        position = [-1,-1,-1]
+        name_reblock_model = self.name + "_reblock"
+        self.reblock_model_attributes(collection)
+        reblock_model = BlockModel(name_reblock_model)
+        DBProvider().clear_collection(name_reblock_model)
+        reblock_collection = DBProvider().select_collection(name_reblock_model)
+        max_coords = self.max_coordenates(collection)
+        x_max =int(max_coords[0])
+        y_max =int(max_coords[1])
+        z_max =int(max_coords[2])
+        counter = 0
+        for x in range(0, x_max, rx):
+            position = [position[0]+ 1, -1, -1]
+            for y in range(0, y_max, ry):
+                position = [position[0], position[1] + 1, -1]
+                for z in range(0, z_max, rz):
+                    position = [position[0], position[1], position[2] + 1]
+                    block = self.resize(collection, x, y, z, rx, ry, rz,position, counter, attributes_types, mass_attribute)
+                    reblock_collection.insert_one(block)
+                    counter += 1
+        return reblock_collection
+
+    def continue_attributes(self, blocks, attribute):
+        value = 0
+        if len(blocks) == 0:
+            return value
+        for block in blocks:
+            if block != None:
+                value += float(block[attribute])
+        return value
+
+    def proportinal_attributes(self, blocks, attribute, mass_attribute):
+        value = 0
+        mass = 0
+        if len(blocks) == 0:
+            return value
+        for block in blocks:
+            if block!= None:
+                value += float(block[attribute]) * float(block[mass_attribute])
+                mass += float(block[mass_attribute])
+        return value/mass
+
+    def categorical_attributes(self, blocks, attribute):
+        attributes = []
+        if len(blocks) == 0:
+            return None
+        for block in blocks:
+            attributes.append(block[attribute])
+        return max(set(attributes), key = attributes.count) 
+
+
+    def max_coordenates(self, collection):
+        max_x = collection.find({}).sort([('x',-1)]).collation(Collation(locale='fr_CA',numericOrdering= True))[0]["x"]
+        max_y = collection.find({}).sort([("y",-1)]).collation(Collation(locale='fr_CA',numericOrdering= True))[0]["y"]
+        max_z = collection.find({}).sort([("z",-1)]).collation(Collation(locale='fr_CA',numericOrdering= True))[0]["z"]
+        return [max_x, max_y, max_z]
+
+
+    def resize(self, collection, x_start, y_start, z_start ,reblock_x, reblock_y, reblock_z, position, counter, attributes_types, mass_attribute):
```

Comment:
This is a confusing name, you should rename it to something like "`create_reblocked_block`" or similar